boot/kernel_os.go

```diff
@@ -109,7 +109,7 @@ func SetNextBoot(s *snap.Info) error {
 	}
 
 	if s.Type != snap.TypeOS && s.Type != snap.TypeKernel && s.Type != snap.TypeBase {
-		return fmt.Errorf("cannot set next boot to snap %q with type %q", s.Name(), s.Type)
+		return fmt.Errorf("cannot set next boot to snap %q with type %q", s.InstanceName(), s.Type)
```

Comment:
This should probably be the actual snap name. We cannot set the boot to _that snap_, no matter what the instance name is.